src/Runtime/python/PyExecutionSessionBase.cpp

```diff
@@ -62,15 +96,24 @@ PyExecutionSessionBase::PyExecutionSessionBase(
 // Run.
 
 std::vector<py::array> PyExecutionSessionBase::pyRun(
-    const std::vector<py::array> &inputsPyArray) {
+    const std::vector<py::array> &inputsPyArray,
+    const std::vector<py::array> &shapesPyArray,
+    const std::vector<py::array> &stridesPyArray) {
   if (!isInitialized)
     throw std::runtime_error(reportInitError());
   if (!_entryPointFunc)
     throw std::runtime_error(reportUndefinedEntryPointIn("run"));
 
   // 1. Process inputs.
   std::vector<OMTensor *> omts;
-  for (auto inputPyArray : inputsPyArray) {
+  assert((inputsPyArray.size() == shapesPyArray.size()) &&
```

Comment:
same here, if the inputs come from the user (which I believe they do), I suggest to have a friendlier error message via throwing an exception.

Reply:
Fixed in a similar way to the above.